src/import/FSHImporter.ts

```diff
@@ -76,41 +81,60 @@ enum Flag {
  * we must call the explicit visitX functions.
  */
 export class FSHImporter extends FSHVisitor {
-  private used = false;
-  private readonly doc: FSHDocument;
+  private currentFile: string;
+  private currentDoc: FSHDocument;
+  private allAliases: Map<string, string>;
 
-  constructor(public readonly file: string = '') {
+  constructor() {
     super();
-    this.doc = new FSHDocument(file);
   }
 
-  visitDoc(ctx: pc.DocContext): FSHDocument {
-    if (this.used) {
-      logger.error('FSHImporter cannot be re-used. Construct a new instance.');
-      return;
-    }
-    this.used = true;
+  import(rawFSHes: RawFSH[]): FSHDocument[] {
+    const docs: FSHDocument[] = [];
+    const contexts: pc.DocContext[] = [];
+    rawFSHes.forEach(rawFSH => {
+      docs.push(new FSHDocument(rawFSH.path));
+      contexts.push(this.parseDoc(rawFSH.content, rawFSH.path));
+    });
+
+    // Import all aliases first
+    this.allAliases = new Map(
+      flatMap(
+        contexts.map((context, index) => {
+          this.currentDoc = docs[index];
+          this.currentFile = this.currentDoc.file ?? '';
+          return this.getAliases(context);
+        }),
+        aliases => Array.from(aliases)
+      )
+    );
+
+    contexts.forEach((context, index) => {
+      this.currentDoc = docs[index];
+      this.currentFile = this.currentDoc.file ?? '';
+      this.visitDoc(context);
+    });
```

Comment:
I doubt it will be a problem, but I would feel better if we reset `this.currentDoc` and `this.currentFile` to `null` here -- since we're done with them.

I'm probably being paranoid, but I've been bitten by something like this before -- where a value stayed in memory from a prior iteration and was used out of context.  Again, it might not even be possible here -- but I'll feel better if we just guard against it. ;-)

Reply:
That's a good idea. I thought about it and initially decided not to, since they are `private`. That said, I can now imagine some situations where someone might try to overwrite it erroneously, since they can't be set as `readonly`. I'll fix it now!